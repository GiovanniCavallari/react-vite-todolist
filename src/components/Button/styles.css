.btn {
    cursor: pointer;
    padding: 12px 16px;
    border-radius: 4px;
    background-color: #aaa;
    color: #fff;
    font-weight: 500;
}

.btn-neutral {
    background-color: #0d6efd;
}

.btn-neutral:hover {
    background-color: #0b5ed7;
}

.btn-success {
    background-color: #198754;
}

.btn-success:hover {
    background-color: #157347;
}

.btn-danger {
    background-color: #dc3545;
}

.btn-danger:hover{
    background-color: #bb2d3b;
}