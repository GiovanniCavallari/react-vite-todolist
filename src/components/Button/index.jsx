import React from 'react';
import P from 'prop-types';

import './styles.css';

function Button({ children, type, onClick }) {
  const types = ['neutral', 'success', 'danger'];
  const buttonType = types.includes(type) ? type : 'neutral';

  return (
    <button
      className={`btn btn-${buttonType}`}
      onClick={onClick}
      data-testid="button"
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  type: P.string,
  onClick: P.func.isRequired,
  children: P.node.isRequired,
};

export default Button;
