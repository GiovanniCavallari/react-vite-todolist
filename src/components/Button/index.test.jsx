import { render } from '@testing-library/react';
import Button from './index';

describe('Test Button Component', () => {
  const onClick = jest.fn();
  const children = 'button';

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const component = <Button onClick={onClick}>{children}</Button>;

    const { asFragment, getByText } = render(component);
    const element = getByText(children);

    expect(asFragment).toMatchSnapshot();
    expect(element.className).toContain('btn-neutral');
  });

  it('should render button with type success', () => {
    const component = (
      <Button type="success" onClick={onClick}>
        {children}
      </Button>
    );

    const { asFragment, getByText } = render(component);
    const element = getByText(children);

    expect(asFragment).toMatchSnapshot();
    expect(element.className).toContain('btn-success');
  });

  it('should render button with type danger', () => {
    const component = (
      <Button type="danger" onClick={onClick}>
        {children}
      </Button>
    );

    const { asFragment, getByText } = render(component);
    const element = getByText(children);

    expect(asFragment).toMatchSnapshot();
    expect(element.className).toContain('btn-danger');
  });

  it('should click on button', () => {
    const component = <Button onClick={onClick}>{children}</Button>;

    const { asFragment, getByText } = render(component);
    const element = getByText(children);

    element.click();

    expect(asFragment).toMatchSnapshot();
    expect(onClick).toBeCalled();
  });
});
