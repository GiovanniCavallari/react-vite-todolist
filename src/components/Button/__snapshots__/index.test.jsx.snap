// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component should click on button 1`] = `[Function]`;

exports[`Test Button Component should render button with type danger 1`] = `[Function]`;

exports[`Test Button Component should render button with type success 1`] = `[Function]`;

exports[`Test Button Component should render correctly 1`] = `[Function]`;
